### DISCUSSION TITLE
Error on building MTconeect agent on Debian Linux

### DISCUSSION THREAD

**Discussion by Varun-Poojary** [mtconnect/cppagent#312] [Q&A]
`apt install libxml2 libxml2-dev cmake git build-essential libcppunit-dev

git clone https://github.com/mtconnect/cppagent.git

cd cppagent

cmake .`

when running cmake . Im getting this error

INFO Shared build: OFF CMake Error at /usr/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:230 (message): Could NOT find Boost (missing: Boost_INCLUDE_DIR) Call Stack (most recent call first): /usr/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE) /usr/share/cmake-3.25/Modules/FindBoost.cmake:2376 (find_package_handle_standard_args) agent_lib/CMakeLists.txt:293 (find_package)

Any Suggestions??

**Comment by wsobel** [comment]
I'd start by reading the section in the README on building the agent on Linux. 

You can also check the Dockerfile to see how to build and create the demo. 

Best - W

**Reply by Varun-Poojary** [reply]
Thank you for your advice. 
I was able to build it successfully.
